addons/website/tests/test_get_current_website.py

```diff
@@ -134,3 +135,26 @@ def test_02_signup_user_website_id(self):
 
         user = self.env['res.users'].create({'website_id': website.id, 'login': 'sad@mail.com', 'name': 'Hope Fully'})
         self.assertTrue(user.website_id == user.partner_id.website_id == website)
+
+    def test_recursive_current_website(self):
+        Website = self.env['website']
+        self.env['ir.rule'].create({
+            'name': 'Recursion Test',
+            'model_id': self.env.ref('website.model_website').id,
+            'domain_force': [(1, '=', 1)],
+            'groups': [],
+        })
+        # Ensure the cache is invalidated, it is not needed at the time but some
+        # code might one day go through get_current_website_id before reaching
+        # this code, making this test useless
+        Website.clear_caches()
```

Comment:
@bso-odoo I pushed a temp commit on the branch, could you have a look? It's changing the test to a simple orm search

About this `clear_caches`, note that it's not really needed, but you can see the need to be sure the test doesn't become silently obsolete later by adding those 2 lines before this line:
```py
        with MockRequest(self.env, website=self.website):
            Website.browse(1).get_current_website()
```

Reply:
Ok.
Note that anything that requires an ACL check would fail, but I'll keep the `.search([])` you suggest.
E.g.
```py
                user = self.env.ref('base.public_user')
                user.with_user(user.id).search([])
```
or
```py
                Website.with_user(self.env.ref('base.public_user').id).get_current_website()
```

Reply:
`Website.with_user(self.env.ref('base.public_user').id).get_current_website()` yes because it's doing a `search()` inside it, so let's just do the search directly ^^